src/pipeline/server.rs

```diff
@@ -11,16 +13,20 @@ use futures::{Future, Poll, Async};
 
 /// A server `Task` that dispatches `Transport` messages to a `Service` using
 /// protocol pipelining.
-pub struct Server<S, T>
-    where S: ServerService,
-          T: Transport,
+pub struct Server<S, T, B>
+    where T: Transport,
+          S: Service<Request = Message<T::Out, Body<T::BodyOut, T::Error>>,
```

Comment:
I would be fine w/ renaming In/Out if we can come up w/ something better. It was mostly copied from finagle terminology.

I think of "in" as messages going into the transport and "out" as messages coming out of the transport. In/Out map to request / response differently depending on if we are talking about a client or server (http/2 makes that even more fuzzy).